### ISSUE TITLE
Cite command creates case sensitivity on name

### ISSUE THREAD

**Issue by BrianSipos** [doxygen/doxygen#10907]
**Describe the bug**
I am using the [`\cite`](https://www.doxygen.nl/manual/commands.html#cmdcite) command from a few places in source markup and in markdown files to reference documents from a BibTeX file.
The BibTeX file uses lower cased keys for its entries.
When the first `\cite` command encountered uses a specific case and matches one of the entries in a case-insensitive way that specific case then becomes the only way that later `\cite` commands will match.

**To Reproduce**
I have a BibTeX file with an entry copied from [rfc9172](https://datatracker.ietf.org/doc/rfc9172/bibtex/) and several uses of `\cite RFC9172` and `\cite rfc9172`.

This causes the following warnings from doxygen and broken links from the produced HTML docs.
The reference cited with upper case key _does_ have a proper link and the entry _is_ present in the Bibliography page.
```
warning: \cite command to 'rfc9172' does not have an associated number
```

**Expected behavior**
If the BibTeX key matching is case-insensitive it should apply to all uses of `\cite` not just the first one.
Otherwise, the matching should be documented as case-sensitive and properly apply to all uses.

**Version**
Doxygen 1.9.1 from package on Ubuntu 22.04.

**Comment by albert-github** [comment]
- Where is it defined that the bibtext references are case independent?
- The doxygen version 1.9.1 is a bit older (January 8, 2021), the current doxygen version is 1.11.0 (though I don't think this will make a difference.

Furthermore to be able to reproduce the problem:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10908

**Comment by BrianSipos** [comment]
I have been scouring documentation at https://www.bibtex.org/ and https://www.ctan.org/pkg/bibtex and https://www.latex-project.org/help/documentation/ and cannot actually find any definition of how the citation key is supposed to be matched. Though all examples I have seen use identical case.

The actual confusing part to me is not that there is case sensitivity (that I could guess) but that the first match from doxygen source is _not_ case sensitive, while later matches are sensitive based on the result of the first match. So for the example below I would expect that only the `xyz` citation would not result in a warning/failure.

I have isolated a minimal test case as [example.zip](https://github.com/doxygen/doxygen/files/15487797/example.zip) which has a single bib reference key `xyz` and citations with different cases. The first citation is accepted and later ones are rejected, even though the last one is the only one which actually matches the bib key by case.

The output of doxygen here is
```
doc.md:1: warning: \cite command to 'xYz' does not have an associated number
doc.md:1: warning: \cite command to 'xyz' does not have an associated number
```

**Comment by albert-github** [comment]
LaTeX itself doesn't like the different cases as well, i.e. when having in a LaTeX file:
```
Test \cite{XYZ} and \cite{xYz} and \cite{xyz}
```
the second and third citation are not resolved either (not really amazing that doxygen has problems in HTML as well).

I think that for doxygen we can relax the restriction regarding the case sensitivity a bit.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10909

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
